system/HTTP/Response.php

```diff
@@ -247,6 +253,7 @@ public function __construct($config)
 		$this->cookiePath     = $config->cookiePath;
 		$this->cookieSecure   = $config->cookieSecure;
 		$this->cookieHTTPOnly = $config->cookieHTTPOnly;
+		$this->cookieSameSite = $config->cookieSameSite ?? 'Lax';
```

Comment:
I think we should also add some check here:
```php
if (! in_array(strtolower($this->cookieSameSite), ['none', 'lax', 'strict'])
{
// throw an exception
}
```

Reply:
The point here made by @samsonasik (https://github.com/codeigniter4/CodeIgniter4/pull/3447#discussion_r465781515) is that since this is a new setting, we need to handle that gracefully when updating using composer. Or adding a note to docs and have a BC discussion ;-)

Reply:
Per RFC, if the value given to samesite does not fall within the three values, it must be treated as `None`.

Reply:
But that's how the browser should handle an illegal attribute. If it is missing, the browser should do... what? I still think CI should be able to not set the attribute at all.

Reply:
> The point here made by @samsonasik (#3447 (comment)) is that since this is a new setting, we need to handle that gracefully when updating using composer.

I believe he was referring to the variable from the config file - and that's true. Here we go with `this->cookieSameSite` as a fallback which is the variable from this class - so we can be sure it's there.

> Per RFC, if the value given to samesite does not fall within the three values, it must be treated as `None`.

Very good point - I missed this note when I was reading RFC yesterday. I guess we should do the same... however I like the idea with  "exception" - it would let developers avoid some simple mistakes.

> But that's how the browser should handle an illegal attribute. If it is missing, the browser should do... what? I still think CI should be able to not set the attribute at all.

All modern browsers have some default value for it. So if it's not set - the browser will add it. IMO there is no point for implementing this feature. This could also potentially leads to certain problems - when a cookie is set in one browser and not in the other.

Reply:
Added exception